modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceReplacementProducts.testcase

```diff
@@ -32,6 +33,66 @@ definition {
 		CommerceConfiguration.commerceTearDown();
 	}
 
+	@description = "COMMERCE-8164 | A discountinued product cannot be replaced by another discontinued product"
+	@priority = "3"
+	test CannotReplaceDiscountProductWithDiscontinuedProduct {
+		task ("Given 2 products are set as Discontinued with a replacement product") {
+			CommerceProducts.openProductsAdmin();
+
+			Search.searchCP(searchTerm = "ABS Sensor");
+
+			CommerceNavigator.gotoTableListEntry(entryName = "ABS Sensor");
+
+			CommerceEntry.gotoMenuTab(menuTab = "SKUs");
+
+			CommerceNavigator.gotoTableListEntry(entryName = "MIN93015");
+
+			SelectFrame(locator1 = "IFrame#IFRAME");
+
+			CommerceProducts.discontinueSKU();
+
+			CommerceProducts.setReplacementSKU(replacementSKU = "MIN55861");
+
+			CommerceProducts.openProductsAdmin();
+
+			Search.searchCP(searchTerm = "U-Joint");
+
+			CommerceNavigator.gotoTableListEntry(entryName = "U-Joint");
+
+			CommerceEntry.gotoMenuTab(menuTab = "SKUs");
+
+			CommerceNavigator.gotoTableListEntry(entryName = "MIN55861");
+
+			SelectFrame(locator1 = "IFrame#IFRAME");
+
+			CommerceProducts.discontinueSKU();
+
+			CommerceProducts.setReplacementSKU(replacementSKU = "MIN93021");
+		}
+
+		task ("and a product is set as Discontinued with a discontinued replacement product") {
+			CommerceProducts.openProductsAdmin();
+
+			Search.searchCP(searchTerm = "Calipers");
+
+			CommerceNavigator.gotoTableListEntry(entryName = "Calipers");
+
+			CommerceEntry.gotoMenuTab(menuTab = "SKUs");
+
+			CommerceNavigator.gotoTableListEntry(entryName = "MIN93021");
+
+			SelectFrame(locator1 = "IFrame#IFRAME");
+
+			CommerceProducts.discontinueSKU();
+
+			CommerceProducts.setReplacementSKU(replacementSKU = "MIN93015");
+		}
+
+		task ("Then the product replacement will not be assigned and an error is shown") {
+			AssertElementPresent(locator1 = "Message#ERROR");
```

Comment:
Let's not end the test here. Add an assert that a product replacement is not assigned. You can navigate back to the product details and use an AssertElementNotPresent for the product replacement field.